### ISSUE TITLE
document how to build this project

### ISSUE THREAD

**Issue by jeffhammond** [kokkos/kokkos-remote-spaces#7]
I am trying to test my MPI changes but cannot figure out what `INCLUDE could not find load file:` means when I compile with GCC 9.3.0 and OpenMP host support.

# Kokkos remote spaces build

```
jrhammon@jrhammon-nuc:~/KOKKOS/remote-spaces/build$ git clean -dfx ; CC=gcc CXX=g++ MPI_CXX=mpig++ cmake .. -DKokkos_DIR=$HOME/KOKKOS/kokkos/build -DKokkos_ENABLE_MPISPACE=ON
warning: failed to remove ./: Invalid argument
Removing ./KokkosRemoteConfig.cmake
Removing ./examples
Removing ./CMakeCache.txt
Removing ./KokkosRemoteConfigVersion.cmake
Removing ./CMakeFiles
-- The CXX compiler identification is GNU 9.3.0
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /home/jrhammon/KOKKOS/kokkos/build/KokkosConfig.cmake:42 (INCLUDE):
  INCLUDE could not find load file:

    /home/jrhammon/KOKKOS/kokkos/build/KokkosTargets.cmake
Call Stack (most recent call first):
  CMakeLists.txt:9 (find_package)


-- Enabled Kokkos devices: OPENMP
-- Found MPI_CXX: /opt/intel/oneapi/mpi/2021.1-beta08/lib/libmpicxx.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Configuring incomplete, errors occurred!
See also "/home/jrhammon/KOKKOS/remote-spaces/build/CMakeFiles/CMakeOutput.log".
```

# Kokkos build

```
jrhammon@jrhammon-nuc:~/KOKKOS/kokkos/build$ git clean -dfx ; CC=gcc CXX=g++ cmake .. -DKokkos_ENABLE_OPENMP=ON && make -j8
warning: failed to remove ./: Invalid argument
-- Setting default Kokkos CXX standard to 11
-- The CXX compiler identification is GNU 9.3.0
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Using -std=gnu++11 for C++11 extensions as feature
-- Execution Spaces:
--     Device Parallel: NONE
--     Host Parallel: OPENMP
--       Host Serial: NONE
-- 
-- Architectures:
-- Found TPLLIBDL: /usr/lib/x86_64-linux-gnu/libdl.so  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jrhammon/KOKKOS/kokkos/build
Scanning dependencies of target kokkoscore
[  4%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_CPUDiscovery.cpp.o
[ 19%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostBarrier.cpp.o
[ 19%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o
[ 19%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o
[ 23%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace.cpp.o
[ 28%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o
[ 33%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o
[ 38%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostThreadTeam.cpp.o
[ 42%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_MemoryPool.cpp.o
[ 47%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_MemorySpace.cpp.o
[ 52%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o
[ 57%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial_Task.cpp.o
[ 61%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o
[ 66%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Spinwait.cpp.o
[ 71%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o
[ 76%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_hwloc.cpp.o
[ 80%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Exec.cpp.o
[ 85%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Task.cpp.o
[ 90%] Linking CXX static library libkokkoscore.a
[ 90%] Built target kokkoscore
Scanning dependencies of target kokkoscontainers
[ 95%] Building CXX object containers/src/CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o
[100%] Linking CXX static library libkokkoscontainers.a
[100%] Built target kokkoscontainers
```

**Comment by jeffhammond** [comment]
On Slack, @jjwilke said he would fix this.

```
$ git clean -dfx ; CC=gcc-10 CXX=g++-10 MPI_CXX=mpicxx cmake .. -DKokkos_DIR=$KOKKOS/install -DKokkos_ENABLE_MPISPACE=ON -DCMAKE_INSTALL_PREFIX=$KOKKOS/install  && make -j4 install
warning: failed to remove ./: Invalid argument
Removing ./CMakeFiles
Removing ./Makefile
Removing ./cmake_install.cmake
Removing ./KokkosRemoteConfigVersion.cmake
Removing ./examples
Removing ./CMakeCache.txt
Removing ./KokkosRemoteConfig.cmake
-- The CXX compiler identification is GNU 10.2.0
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/bin/g++-10 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Enabled Kokkos devices: OPENMP
-- Found MPI_CXX: /usr/local/Cellar/open-mpi/4.0.4_1/lib/libmpi.dylib (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Configuring done
-- Generating done
-- Build files have been written to: $KOKKOS/remote-spaces/build
Scanning dependencies of target kokkosremote
[ 16%] Building CXX object CMakeFiles/kokkosremote.dir/src/MPISPACE/Kokkos_MPISpace.cpp.o
```

**Comment by jjwilke** [comment]
Addressed by PR #14, which is now merged into main.
